strawberryfields/parameters.py

```diff
@@ -19,335 +19,334 @@
+    A MeasuredParameter instance, given as a parameter to a
+    :class:`~strawberryfields.ops.Operation` constructor, represents
+    a dependence of the Operation on classical information obtained by
+    measuring a subsystem of the register.
 
-        Returns:
-          Parameter: x as a Parameter instance
-        """
-        if isinstance(x, Parameter):
-            return x
-        return Parameter(x)
-
-    # the arithmetic methods below basically are just responsible for calling _unwrap_and_cast which exposes self.x, then the arithmetic ops of the supported parameter types take over
-    def __add__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(temp + other)
-
-    def __radd__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(other + temp)
-
-    def __sub__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(temp - other)
-
-    def __rsub__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(other - temp)
-
-    def __mul__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(temp * other)
-
-    def __rmul__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(other * temp)
-
-    def __truediv__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(temp / other)
-
-    def __rtruediv__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(other / temp)
-
-    def __pow__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(temp ** other)
-
-    def __rpow__(self, other):
-        temp, other = self._unwrap_and_cast(other)
-        return self._wrap(other ** temp)
-
-    def __neg__(self):
-        # pylint: disable=invalid-unary-operand-type
-        return Parameter(-self.x)
-
-    # other properties
-    @property
-    def shape(self):
-        """Returns the shape of array parameters."""
-        try:
-            return self.x.shape
-        except AttributeError:
-            return None
-
-    # comparisons
-    def __eq__(self, other):
-        """Equality comparison.
-
-        .. note:: This method may be too permissive, maybe it should return False if either parameter is not a numbers.Number or a np.ndarray?
+    Used for deferred measurements, i.e., using a measurement's value
+    symbolically in defining a gate before the numeric value of that
+    measurement is available.
+
+    Former RegRefTransform functionality is provided by the sympy.Symbol base class.
+
+    Args:
+        regref (RegRef): register reference responsible for storing the measurement result
+    """
+
+    def __new__(cls, regref):
+        # sympy.Basic.__new__ wants a name, other arguments must not end up in self._args
+        return super().__new__(cls, 'q'+str(regref.ind))
```

Comment:
I guess we're forced to pick a convention here, even though the user is not forced to use ``q``?

Reply:
Ahhh, not exactly. As far as I can tell this is just an internal name used by sympy, which could be anything. The name  used when doing a string conversion is defined in the `_sympystr` method below.